airflow/models.py

```diff
@@ -679,6 +682,10 @@ def are_dependencies_met(
                         case([(TI.state == State.SUCCESS, 1)], else_=0)),
                     func.sum(
                         case([(TI.state == State.SKIPPED, 1)], else_=0)),
+                    func.sum(
```

Comment:
Each ``func.sum(...)`` statement should be wrapped in ``func.coalesce(func.sum(...), 0)`` for Py3 compatibility. Otherwise ``func.sum()`` can return None if there are no rows, and that might create type errors because these results are treated as ints later.